src/shared/PublicKeyString.ts

```diff
@@ -0,0 +1,52 @@
+import assert from '../utils/assert';
+import { PublicKey } from '@solana/web3.js';
+
+/**
+ * Synonym for `string` to differentiate strings that should contain a base58 representation of
+ * a {@link PublicKey}
+ *
+ * @private
+ */
+export type PublicKeyString = string;
+
+/**
+ * Checks if a string is valid base58 Solana via a Regex.
+ * @private
+ */
+export function isValidSolanaAddress(address: string) {
+  return /^[0-9a-zA-Z]{43,88}$/.test(address);
+}
+
+/**
+ * Checks if a string is valid PublicKey address.
+ * @private
+ */
+export function isValidPublicKeyAddress(address: string) {
```

Comment:
I don't like the naming here `isValidSolanaAddress` is a paraphrase of `isValidPublicKeyAddress`.

`isValidSolanaAddress` should be more explicit like `isBase58Address` even though it doesn't really fully test for that either.

IMO this logic is redundant anyway since the `PublicKey` constructor does that for us and the try/catch is more than enough.